### PR TITLE
ENT-3800 Prevent systemv script from managing processes inside other containers

### PR DESCRIPTION
**Pull request by nickanderson**

### CODE REVIEW

File: misc/init.d/cfengine3.in [hunk 1]
```diff
@@ -182,7 +182,7 @@ cf_get_matching_pids()
 
     if [ $INSIDE_CONTAINER = -1 ]; then
         # Are we inside a container?
-        if [ -f /proc/self/cgroup ] && egrep '[0-9]+:devices:/[^:]' /proc/self/cgroup > /dev/null; then
+        if [ -f /proc/$pid/cgroup ] && sed 's_/\(-\|system\|user\).slice__;s_/user-[0-9]*.slice__;s_/session-[0-9]*.scope__;s_/cfengine3.service__' /proc/$pid/cgroup | egrep '[0-9]+:devices:/[^:]' > /dev/null; then
```
Comment (Lex-2008) [inline misc/init.d/cfengine3.in]: I think **both** entries of `$pid` should be replaced with `self` here

Comment (nickanderson) [reply misc/init.d/cfengine3.in]: yeah i see i had a bad diff i go tit., will fix and re-submit

**Comment by nickanderson** [comment]
@Lex-2008  Updated

**Comment by nickanderson** [comment]
ping @Lex-2008

**Pull request merged**